miden-lib/asm/kernels/transaction/lib/prologue.masm

```diff
@@ -368,6 +368,7 @@ proc.validate_new_account
             # empty SMT)
             exec.constants::get_empty_smt_root assert_eqw.err=ERR_PROLOGUE_NEW_NON_FUNGIBLE_FAUCET_INVALID_RESERVED_SLOT
             # => []
+            push.1001 drop                              # TODO: remove line, see miden-vm/#1122
```

Comment:
Why do we need to add this here specifically? Did we have an error with decorators previously?

Reply:
Yeah, it's what Bobbin mentioned yesterday (and related to the error mentioned in the comment). Essentially for asserts with errors the MAST root would be the same and thus the error returned by the VM could be unrelated to the actual error happening in the execution. I do not know the inner workings but adding this push and drop fixes it and we can assert for the actual error in the test.